certsuite/harness.py

```diff
@@ -451,6 +453,8 @@ def run_tests(args, config):
     finally:
         if output_zipfile:
             print >> sys.stderr, "Results saved to %s" % output_zipfile
+        if remove_after_mcts:
+            marionette_install.uninstall()
```

Comment:
The patch is attempted to remove the marionette-extension we installed if we found there's no marionette available in the device under testing.

If the device is flashed with user build (no marionette  inside),
```
marionette_install.install()
```
will install the marionette-extension for testing.
If the device is flashed with engineer build. The uninstall function will remove nothing since the INSTALL_DIR doesn't point to the dictionary that marionette is and won't raise an error. I believe we can simply ignore the case if it's an engineer build.

https://github.com/mozilla-b2g/marionette-extension/blob/master/marionette_extension/installer.py#L66